subsys/net/lib/dns/Kconfig

```diff
@@ -194,6 +194,17 @@ config DNS_RESOLVER_CACHE_MAX_ENTRIES
 
 endif # DNS_RESOLVER_CACHE
 
+config DNS_RESOLVER_PACKET_FORWARDING
+	bool "Forwards received DNS packets to application"
+	depends on !DNS_RESOLVER_CACHE
+	help
+	  This allows forwarding of received packets from DNS servers
+	  to applications that register the forwarding callback.
+	  Application must store query's ID that has to be passed to
+	  DNS resolver and then update the received packet from resolver
+	  with the stored ID, as a DNS response has to have the same query ID
+	  as the query that triggered it.
```

Comment:
This part souds a bit cryptic to me, what exactly does it mean that the application needs to update the response packet with query ID?